dts/xtensa/intel/intel_adsp_ace15_mtpm.dtsi

```diff
@@ -0,0 +1,126 @@
+/*
+ * Copyright (c) 2022 Intel Corporation
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <xtensa/xtensa.dtsi>
+#include <mem.h>
+
+/ {
+	cpus {
+		#address-cells = <1>;
+		#size-cells = <0>;
+
+		cpu0: cpu@0 {
+			device_type = "cpu";
+			compatible = "cdns,tensilica-xtensa-lx7";
+			reg = <0>;
+			cpu-power-states = <&idle &suspend &off>;
+		};
+
+		cpu1: cpu@1 {
+			device_type = "cpu";
+			compatible = "cdns,tensilica-xtensa-lx7";
+			reg = <1>;
+			cpu-power-states = <&idle &suspend &off>;
+		};
+
+		cpu2: cpu@2 {
+			device_type = "cpu";
+			compatible = "cdns,tensilica-xtensa-lx7";
+			reg = <2>;
+			cpu-power-states = <&idle &suspend &off>;
+		};
+	};
+
+	power-states {
+		idle: idle {
+			compatible = "zephyr,power-state";
+			power-state-name = "runtime-idle";
+			min-residency-us = <0>;
+			exit-latency-us = <0>;
+		};
+		suspend: suspend {
+			compatible = "zephyr,power-state";
+			power-state-name = "suspend-to-idle";
+			min-residency-us = <200>;
+			exit-latency-us = <100>;
+		};
+		/* PM_STATE_SOFT_OFF can be entered only by calling pm_state_force.
+		 * The procedure is triggered by IPC from the HOST (SET_DX).
+		 */
+		off: off {
+			compatible = "zephyr,power-state";
+			power-state-name = "soft-off";
+			min-residency-us = <2147483647>;
+			exit-latency-us = <0>;
+		};
+	};
+
+	sram0: memory@a0020000 {
+		device_type = "memory";
+		compatible = "mmio-sram";
+		reg = <0xa0020000 DT_SIZE_K(2816)>;
+	};
+
+	sram1: memory@a0000000 {
+		device_type = "memory";
+		compatible = "mmio-sram";
+		reg = <0xa0000000 DT_SIZE_K(60)>;
+	};
+
+	reserverd_mem: reserved-memory {
```

Comment:
This is a new feature and I'll admit I'm not 100% sure of how it's supposed to be used (@carlocaione might have input).  But this seems wrong to me.  This is for memory that is provided by the platform but not available to "regular" Zephyr usage, so it has its own driver interface.

In fact IMR isn't "reserved" to the linker script.  Zephyr can place regular C data and code there with __imr/__imrdata attributes.  This probably wants to be a regular RAM block symmetric with the existing ones for hp/lp-sram.

Reply:
`reserved-memory` support as compatible was removed with #45030.

Reply:
Ah, OK.  See, this is why the ability to pull in upstream subsystem authors during review is valuable.  I saw this and actually thought it was the new feature and not the old one because I haven't been paying enough attention. :)